remoteappmanager/rest/resource.py

```diff
@@ -0,0 +1,166 @@
+from tornado import gen
+
+
+class Resource:
+    """Base class for REST resources.
+    To implement a new Resource class, inherit from this subclass
+    and reimplement the CRUD class methods with the appropriate
+    logic.
+    """
+    @classmethod
+    @gen.coroutine
+    def create(cls, representation):
+        """Called to create a resource with a given representation
+        The representation is a dictionary containing keys. The
+        reimplementing code is responsible for checking the validity
+        of the representation.
+
+        Parameters
+        ----------
+        representation: dict
+            A dictionary containing the representation as from the
+            HTTP request.
+
+        Returns
+        -------
+        id: str
+            An identifier identifying the newly created resource.
+            It must be unique within the collection.
+
+        Raises
+        ------
+        UnprocessableRepresentation:
+            Raised when the representation does not validate according
+            to the resource expected representation.
+        NotImplementedError:
+            If the resource does not support the method.
+        """
+        raise NotImplementedError()
+
+    @classmethod
+    @gen.coroutine
+    def retrieve(cls, identifier):
+        """Called to retrieve a specific resource given its
+        identifier.
+
+        Parameters
+        ----------
+        identifier: str
+            A string identifying the resource
+
+        Returns
+        -------
+        representation: dict
+            a dict representation of the resource.
```

Comment:
The representation is written directly (without modification) to the response body.